packages/shared/src/components/squads/SquadMemberMenu.tsx

```diff
@@ -1,51 +1,158 @@
+> = {
+  [MenuItemTitle.AddAsOwner]: (memberName, squadName) =>
+    `${memberName} will get the same permissions as you have. Adding as co-owner will not replace you as the owner of ${squadName}`,
+  [MenuItemTitle.PromoteToModerator]: () => '', // TODO: awaiting product's response
+  [MenuItemTitle.DemoteToModerator]: (memberName) =>
+    `${memberName} will no longer have owner permissions. You can always reverse this decision later`,
+  [MenuItemTitle.DemoteToMember]: () => '', // TODO: awaiting product's response
+  [MenuItemTitle.RemoveMember]: (memberName, squadName) =>
+    `${memberName} will be a Blocked members and will no longer have access to ${squadName}. They will not be able to rejoin unless you unblock them.`,
+};
 
 export default function SquadMemberMenu({
-  squadId,
-  memberId,
-}: {
-  squadId: string;
-  memberId: string;
-}): ReactElement {
+  squad,
+  member,
+}: SquadMemberMenuProps): ReactElement {
   const { user } = useContext(AuthContext);
+  const { onUpdateRole } = useSquadActions();
+  const { showPrompt } = usePrompt();
+  const onUpdateMember = async (
+    role: SourceMemberRole,
+    title: MenuItemTitle,
+  ) => {
+    const hasConfirmed = await showPrompt({
+      title: `${title}?`,
+      description: promptDescription[title](member.user.name, squad.name),
+      okButton: { title, className: 'btn-primary-ketchup' },
+      content: <UserShortInfo user={member.user} />,
+      promptSize: ModalSize.Small,
+      className: { buttons: 'mt-6' },
+    });
+
+    if (hasConfirmed) {
+      onUpdateRole({ sourceId: squad.id, memberId: member.user.id, role });
+    }
+  };
+
+  const options: ContextMenuItemProps[] = useMemo(() => {
+    if (!member) return [];
+
+    const getUpdateRoleFn =
+      (role: SourceMemberRole, title: MenuItemTitle) => () =>
+        onUpdateMember(role, title);
+    const menu: ContextMenuItemProps[] = [];
+    const promoteToOwner = {
+      text: MenuItemTitle.AddAsOwner,
+      icon: <StarIcon size={IconSize.Small} />,
+      action: getUpdateRoleFn(SourceMemberRole.Owner, MenuItemTitle.AddAsOwner),
+    };
+    const promoteToModerator = {
+      text: MenuItemTitle.DemoteToModerator,
+      icon: <UserIcon size={IconSize.Small} />,
+      action: getUpdateRoleFn(
+        SourceMemberRole.Moderator,
+        MenuItemTitle.PromoteToModerator,
+      ),
+    };
+    const demoteToModerator = {
+      text: MenuItemTitle.DemoteToModerator,
+      icon: <BlockIcon size={IconSize.Small} />,
+      action: getUpdateRoleFn(
+        SourceMemberRole.Moderator,
+        MenuItemTitle.DemoteToModerator,
+      ),
+    };
+    const demoteToMember = {
+      text: MenuItemTitle.DemoteToMember,
+      icon: <SquadIcon size={IconSize.Small} />,
+      action: getUpdateRoleFn(
+        SourceMemberRole.Member,
+        MenuItemTitle.DemoteToMember,
+      ),
+    };
+
+    const canUpdateRole = squad.currentMember?.permissions.includes(
```

Comment:
Maybe we can extract `permissions.includes` into utility function, I see even now being called on few places, it will make more sense if we add additional logic also. We have similar utility functions on the API side.

Reply:
Do you mean the arrays? I think if we re-create the one we have in API, it might lead to discrepancies and more maintenance work. Though I agree, checking in FE is a bit verbose now, we need to come up with something to lessen it. I'll try to create some functions.

Reply:
Yeah, array iteration for now, API was just an example to where we do have those kind of utility functions. 👍